### PR TITLE
AMQP messages for third party trading engine integration

### PR DESCRIPTION
**Pull request by ysv**
closed #2213

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -129,6 +129,21 @@ def to_matching_attributes
...
 
+  def as_json_for_events_processor
+    { id: id,
+      member_id: member_id,
+      ask:       ask,
+      bid:       bid,
+      type:      type,
+      ord_type:  ord_type,
+      price:     price,
+      volume:    volume,
```
Comment (dnfd) [inline app/models/order.rb]: Would be great to get `uid` in order payload.

File: lib/daemons/account.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    if liabilities_was != liabilities_number
+      [
+        "Liabilities number was #{liabilities_was} and now it's #{liabilities_number}",
+        "Recalculating member balances"
+      ].join("\n").tap { |m| Rails.logger.info m }
+      liabilities_was = liabilities_number
+      break
+    end
+    [
+      "Liabilities number didn't change since last update",
```
Comment (dnfd) [inline lib/daemons/account.rb]: It produces a lot of "empty" logs, hard to find useful info while debugging

File: lib/daemons/account.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      type:          :liability,
+      kind:          :locked,
+      currency_type: currency.type
+    ).code
+
+    Account.where(currency: currency).find_in_batches do |accounts_group|
+      ActiveRecord::Base.transaction do
+        accounts_group.each do |legacy_account|
+          balance = Operations::Liability.where(code:      main_liability_code,
+                                                member_id: legacy_account.member_id)
```
Comment (dnfd) [inline lib/daemons/account.rb]: `currency: currency.id,`

File: lib/daemons/account.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                                         .sum('credit - debit')
+
+          locked = Operations::Liability.where(code:      locked_liability_code,
+                                               member_id: legacy_account.member_id)
```
Comment (dnfd) [inline lib/daemons/account.rb]: `currency: currency.id,`

**Review by dnfd — CHANGES_REQUESTED** [review]

File: app/models/order.rb [hunk 5]
```diff
@@ -129,6 +129,22 @@ def to_matching_attributes
...
+  def as_json_for_events_processor
+    { id:         id,
+      member_id:  member_id,
+      member_uid: member.uid,
+      ask:        ask,
+      bid:        bid,
+      type:       type,
+      ord_type:   ord_type,
+      price:      price,
+      volume:     volume,
```
Comment (dnfd) [inline app/models/order.rb]: Add origin_volume, please :)

**Comment by ysv** [comment]
account.rb SQL query example
```
Operations::Account Load (0.6ms)  SELECT  `operations_accounts`.* FROM `operations_accounts` WHERE `operations_accounts`.`type` = 'liability' AND `operations_accounts`.`kind` = 'main' AND `operations_accounts`.`currency_type` = 'fiat' LIMIT 1
  ↳ lib/daemons/account.rb:36
  Operations::Account Load (0.4ms)  SELECT  `operations_accounts`.* FROM `operations_accounts` WHERE `operations_accounts`.`type` = 'liability' AND `operations_accounts`.`kind` = 'locked' AND `operations_accounts`.`currency_type` = 'fiat' LIMIT 1
  ↳ lib/daemons/account.rb:42
  Account Load (0.3ms)  SELECT  `accounts`.* FROM `accounts` WHERE `accounts`.`currency_id` = 'usd' ORDER BY `accounts`.`id` ASC LIMIT 1000
  ↳ lib/daemons/account.rb:48
   (0.2ms)  BEGIN
  ↳ lib/daemons/account.rb:49
   (0.3ms)  SELECT SUM(credit - debit) FROM `liabilities` WHERE `liabilities`.`code` = 201 AND `liabilities`.`member_id` = 1 AND `liabilities`.`currency_id` = 'usd'
  ↳ lib/daemons/account.rb:54
   (0.3ms)  SELECT SUM(credit - debit) FROM `liabilities` WHERE `liabilities`.`code` = 211 AND `liabilities`.`member_id` = 1 AND `liabilities`.`currency_id` = 'usd'
  ↳ lib/daemons/account.rb:59
   (0.2ms)  COMMIT
```

**Pull request merged**